frontend/src/container/ExplorerOptions/ExplorerOptions.tsx

```diff
@@ -71,6 +77,10 @@ function ExplorerOptions({
 		setIsSaveModalOpen(false);
 	};
 
+	const { role } = useSelector<AppState, AppReducer>((state) => state.app);
+
+	const allowedRoles = [USER_ROLES.ADMIN, USER_ROLES.AUTHOR, USER_ROLES.EDITOR];
```

Comment:
Can we move this outside the function, so that it will not get created every time the state change?

Reply:
done